worker/src/websocket.rs

```diff
@@ -197,6 +197,24 @@ impl WebSocket {
             closures: Some((message_closure, error_closure, close_closure)),
         })
     }
+
+    pub fn serialize_attachment<T: Serialize>(&self, value: T) -> Result<()> {
+        self.socket
+            .serialize_attachment(serde_wasm_bindgen::to_value(&value)?)
+            .map_err(Error::from)
+    }
```

Comment:
I used the `Storage` get / set API as prior art for this, let me know if there's a better approach. 

If I understand this correctly, it converts it to a string before it gets sent to the JS serialize attachment method, meaning it gets "double serialised". The more "correct" apporach might be to have the trait bound to ensure it meets the "Structural Copy" requirement, but then just pass the struct to JS as is for serialization? 

Not sure how serialization would work though. So perhaps the double serialization is the best way to go.